cli/medperf/entities/interface.py

```diff
@@ -52,3 +52,10 @@ def write(self) -> str:
         Returns:
             str: Path to the stored entity
         """
+
+    @property
+    def identifier(self):
+        return self.id or self.generated_uid
+
+    def is_registered(self):
```

Comment:
Loving this! I think this is the kind of cases where implementing the method (or property) inside the abstract class is valid. One thing to think about this is we're indicating we expect `self.id` and `self.generated_uid` to exist in any implementatino of the abstract method. It would be nice to find a way to make this assumption more explicit. (maybe have the interface also inherit from the MedperfSchema later on).

I think we can also have `is_registered` as a property

Reply:
Sounds good regarding the schema inheritance! It is better to define the "common" fields explicitly.

And yes! is_registered can be a property. Will do that